src/app/container/deregister-modal/deregister-modal.component.ts

```diff
@@ -27,28 +28,62 @@ import { RegisterToolService } from './../register-tool/register-tool.service';
   styleUrls: ['./deregister-modal.component.css']
 })
 export class ModalComponent implements OnInit {
+  @Input() toolTipMessage: string;
   @Input() refreshMessage: boolean;
+  @Input() option: string;
+  @Input() organizationID: number;
+  @Input() organizationStatus: string;
+  @Input() organizationName: string;
 
-  constructor(private registerToolService: RegisterToolService, private confirmationDialogService: ConfirmationDialogService) {}
+  constructor(
+    private registerToolService: RegisterToolService,
+    private confirmationDialogService: ConfirmationDialogService,
+    private requestsService: RequestsService
+  ) {}
 
   ngOnInit() {}
-  deregister() {
+
+  deregisterTool() {
     this.registerToolService.deregisterTool();
   }
 
+  deregisterOrganization(id: number) {
+    this.requestsService.deleteOrganization(id);
+  }
+
   openDeleteDialog() {
-    const confirmationDialogData: ConfirmationDialogData = {
+    const toolDialogData: ConfirmationDialogData = {
       title: 'Are you sure you wish to delete this tool?',
       message: `All information associated with this tool will be deleted.`,
       cancelButtonText: 'Cancel',
       confirmationButtonText: 'Delete'
     };
+
+    const organizationDialogData: ConfirmationDialogData = {
```

Comment:
It doesn't feel right to have this here. This component is for something else, not deleting organizations.

Reply:
Implemented an additional input that takes in the dialog data and moved the two existing dialogDatas to their respectful component files.